tartufo/scanner.py

```diff
@@ -349,19 +349,17 @@ def config_data(self, data: MutableMapping[str, Any]) -> None:
         self._config_data = data
 
     @cached_property
-    def excluded_signatures(self) -> tuple:
-        if len(self._excluded_signatures) == 0:
-            signatures = list(self.global_options.exclude_signatures or ())
-            exclude_signatures = list(self.config_data.get("exclude_signatures", ()))
+    def excluded_signatures(self) -> Tuple[str, ...]:
+        if self._excluded_signatures is None:
+            signatures = (tuple(self.global_options.exclude_signatures) or ()) + (
+                tuple(self.config_data.get("exclude_signatures", ()))
+            )
             try:
-                signatures = signatures + [
-                    x for x in exclude_signatures if x not in signatures
-                ]
-                signatures = [
+                signatures = tuple(
                     signature["signature"]
-                    for signature in cast(List[Dict[str, str]], signatures)
-                ]
-                self._excluded_signatures = tuple(cast(List[str], signatures))
+                    for signature in cast(Tuple[Dict[str, str]], signatures)
+                )
+                self._excluded_signatures = tuple(set(signatures))
```

Comment:
This makes no sense to me. Why use a tuple comprehension, and then convert it to a set, and then convert it back to a tuple -- instead of starting with a set comprehension and then converting it once? If we're worried about mutating the type of `signatures`, then just eliminate the intermediate variable and store directly into the destination.

Reply:
Yeah... I had very similar thoughts. And had your words echoing in my head as I did it. I just didn't want to have to think about all the other places that might be depending on a tuple vs a set. In theory it shouldn't matter. But... you never know.

Reply:
@rbailey-godaddy Since it's already getting to dinner time, I'm happy to leave this stewing overnight in case you want to take a moment in the morning to un-thrashify this code. 😄

Reply:
But, also want to keep in mind that we're more than a week overdue for 3.0 now, so... limiting scope is good.